cumulus_etl/formats/deltalake.py

```diff
@@ -93,38 +93,33 @@ def _write_one_batch(self, batch: Batch) -> None:
         delta_table.generate("symlink_format_manifest")
 
     def update_delta_table(self, updates: pyspark.sql.DataFrame, groups: set[str]) -> delta.DeltaTable:
-        full_path = self._table_path(self.dbname)
-
-        try:
-            # Load table -- this will trigger an AnalysisException if the table doesn't exist yet
-            table = delta.DeltaTable.forPath(self.spark, full_path)
-
-            # Determine merge condition
-            conditions = [f"table.{field} = updates.{field}" for field in self.uniqueness_fields]
-            condition = " AND ".join(conditions)
+        table = (
+            delta.DeltaTable.createIfNotExists(self.spark)
+            .addColumns(updates.schema)
+            .clusterBy(*self.uniqueness_fields)
+            .location(self._table_path(self.dbname))
+            .execute()
+        )
```

Comment:
This is the interesting new bit. The rest of the delta is unindenting and removing the except handler that created the table.

Now, we just create/get the table in the same go. Using the delta lake specific method was the only way I could find to specify clustering - and it's a cleaner python code path.